### PR TITLE
[Parser] Add third-party parser package

### LINKED ISSUE #1127: [Parser] Enable TiDB Parser
Part of #1126

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #1127.

Code structure: All third party parser related code will live at `pkg/sql/tpp`. The `tpp` package exposes a function `ParseAndSplit(driver, sql string)` will be called by `pkg/sql/split.go`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,10 +10,11 @@ require (
 	github.com/mattn/go-runewidth v0.0.6 // indirect
 	github.com/mattn/go-sqlite3 v1.11.0
 	github.com/olekukonko/tablewriter v0.0.1
+	github.com/pingcap/parser v0.0.0-20190613045206-37cc370a20a4
+	github.com/pingcap/tidb v0.0.0-20190625145607-60965b006877
```
Comment (wangkuiyi) [inline go.mod:14]: Should we fix the version of TiDB as something like [`v3.0.5`](https://github.com/pingcap/tidb/tree/v3.0.5)?

File: pkg/sql/split.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// Substitute this function for real SQLFlow parser later.
+func extendedSyntaxParse(sql string) (string, error) {
+	extendedSyntax := "to train;"
+	if strings.HasPrefix(sql, extendedSyntax) {
+		return extendedSyntax, nil
+	}
+	return "", fmt.Errorf("SQLFlow parser error %v", sql)
+}
+
+func split(driver, sql string) ([]string, error) {
```
Comment (wangkuiyi) [inline pkg/sql/split.go]: driver => sql_engine or dbms

Comment (wangkuiyi) [inline pkg/sql/split.go]: split => parse

Comment (wangkuiyi) [reply pkg/sql/split.go]: parse(dbms, sql_program string)

File: pkg/sql/split.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
```
Comment (wangkuiyi) [inline pkg/sql/split.go]: split.go => parse.go

File: pkg/sql/split_test.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
```
Comment (wangkuiyi) [inline pkg/sql/split_test.go]: split_test.go => parse_test.go

File: pkg/sql/tpp/README.md [hunk 5]
```diff
@@ -69,93 +69,63 @@ It is notable that MySQL doesn't provide sufficient documentation on how to call
...
-  string sql = 1;
-  string extension = 2;
-  string error = 3;
-}
-
-service Parser {
-  rpc Parse (ParserRequest) returns (ParserResponse) {}
+```
+type ParseResult struct {
+	SQL      []string `json:"sql"`
```
Comment (wangkuiyi) [inline pkg/sql/tpp/README.md:76]: Do we need to dump the ParseResult into a JSON?

File: pkg/sql/tpp/README.md [hunk 6]
```diff
@@ -69,93 +69,63 @@ It is notable that MySQL doesn't provide sufficient documentation on how to call
...
-    i, e := external_parser(sql)
-    if e != nil {
-      return e
-    }
-    
-    if i != -1 {
-      return SQLFlow.Parse(sql[i,:]) // Parse the right part.
-    }
-    return nil
+func Parse(sql_program string) (nodes, error) {
```
Comment (wangkuiyi) [inline pkg/sql/tpp/README.md:99]: Where is the implementation of this `Parse` function in the Go source files?

File: pkg/sql/tpp/tpp.go [hunk 7]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
```
Comment (wangkuiyi) [inline pkg/sql/tpp/tpp.go:1]: tpp => third_party_parser or native_parser

File: pkg/sql/split.go (hunk 3 above)
Comment (tonyyang-svail) [reply pkg/sql/split.go]: Done.

File: pkg/sql/split.go (hunk 2 above)
Comment (tonyyang-svail) [reply pkg/sql/split.go]: Done.

Comment (tonyyang-svail) [reply pkg/sql/split.go]: Done.

File: pkg/sql/split_test.go (hunk 4 above)
Comment (tonyyang-svail) [reply pkg/sql/split_test.go]: Done.

File: pkg/sql/tpp/README.md (hunk 5 above)
Comment (tonyyang-svail) [reply pkg/sql/tpp/README.md:76]: Sorry for the confusion. I have updated the documentation. Basically, `sql_engine.Parse` will invoke a Java command-line which takes a SQL string and outputs the parsing result in JSON format. Then SQLFlow can parse the output into the following Go struct.

File: pkg/sql/tpp/README.md (hunk 6 above)
Comment (tonyyang-svail) [reply pkg/sql/tpp/README.md:99]: `pkg/sql/parse.go`.

File: pkg/sql/tpp/tpp.go (hunk 7 above)
Comment (tonyyang-svail) [reply pkg/sql/tpp/tpp.go:1]: go lint forbids using an underscore in the package name. And the package name is better to be consistent with the directory name. So I am keeping the `tpp`.

```
hookid: go-lint

pkg/sql/third_party_parser/third_party_parser.go:14:1: don't use an underscore in package name
Found 1 lint suggestions; failing.
pkg/sql/third_party_parser/tidb_parser.go:14:1: don't use an underscore in package name
Found 1 lint suggestions; failing.
pkg/sql/third_party_parser/tidb_parser_test.go:14:1: don't use an underscore in package name
Found 1 lint suggestions; failing.
```

File: go.mod (hunk 1 above)
Comment (tonyyang-svail) [reply go.mod:14]: This line is generated by running `go get github.com/pingcap/tidb@v3.0.0`. I am not sure why it appears to be `v.0.0.0-2019....` in the `go.mod`.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**